### PR TITLE
Improve drone name extraction for UI display

### PR DESCRIPTION
**Pull request by gsfr**
@ryansanford Please merge prior to bug bash and reset dev instance to reflect improved bootstrapping results.

Also, the referenced reaper commit is what should be the latest bali tag in scitran/reaper.

### CODE REVIEW

**Comment by nagem** [comment]
LGTM

**Comment by gsfr** [comment]
*Importing* seems to work, although the UI isn't quite right yet. Looks like the UI is currently rendering the `id` field, rather than the `name` field.

```
origin: {
    method: "importer",
    type: "device",
    id: "importer_Admin Import",
    name: "Admin Import"
}
```

Reaping also looks good, but same display problem.

```
origin: {
    method: "reaper",
    type: "device",
    id: "reaper_cnimr",
    name: "cnimr"
}
```

**Comment by nagem** [comment]
I'll open a ticket on the frontend to use the new fields.

**Comment by gsfr** [comment]
Looks like we're getting some errors from our cron jobs:
```
uwsgi_1          | 2016-04-07 23:58:01      scitran.api                  base.py   305:WARN None 400 X-SciTran-Method or X-SciTran-Name header missing
uwsgi_1          | [pid: 17|app: 0|req: 7/9] 172.17.0.9 () {34 vars in 397 bytes} [Thu Apr  7 23:58:01 2016] POST /api/jobs/reap => generated 84 bytes in 0 msecs (HTTP/1.1 400) 3 headers in 120 bytes (2 switches on core 1)
```

**Comment by nagem** [comment]
Talked with Ryan, opening up a ticket in fly/fly.

**Comment by gsfr** [comment]
Rebased

**Comment by nagem** [comment]
Pending review from @gsfr or @kofalt of the DB upgrade script, this is good to go.

**Comment by gsfr** [comment]
The DB updates are not atomic. That may or may not be a problem depending on how we apply them. Could our DB be modified by other processes while the updates are being applied? Ping @ryansanford.

**Comment by ryansanford** [comment]
@nagem @gsfr Flywheel dev environment and production scripts stop all containers before checking and upgrading the schema. This protects the main-line use cases only.

**Comment by nagem** [comment]
And unfortunately we'd need [this ticket](https://jira.mongodb.org/browse/SERVER-1243) to find and update all the file origins in the file array in one go.

File: api/base.py [hunk 1]
```diff
@@ -182,9 +180,15 @@ def set_origin(self, drone_request, drone_name):
                 'id': self.uid
             }
         elif drone_request:
+
+            method = self.request.headers.get('X-SciTran-Method')
+            name = self.request.headers.get('X-SciTran-Name')
```
Comment (kofalt) [inline api/base.py]: Propose that we add `.strip()` to the end of each of these to remove whitespace.

**Comment by kofalt** [comment]
LGTM.

Non-atomic schema upgrades is the way to go - frequently, those run without locks or durability as well :) It should be abundantly clear that serving requests in halfway-house mode has undefined behaviour.

Maybe a comment to that effect at the top of the script would be wise?

**Pull request merged**